### PR TITLE
This PR includes all config required for running unit test using Tekton-as-a-code

### PR DESCRIPTION
**Pull request by rupalibehera**
After this PR is merged running unit tests on nightly PR will be possible.

### CODE REVIEW

**Comment by rupalibehera** [comment]
@chmouel I used all your pipeline scripts for inspiration, I have not tested it but raising the PR for initial review to check if I understood everything correctly.

**Comment by chmouel** [comment]
Seems like we have some issues with owner validation, altho you seems to have done things right. let me see if i can hae a look at this later tomorrow...

**Comment by chmouel** [comment]
Oh yeah i forgot that, we don't allow modify the owners on PR, unless we are `already` owners of the repo for security reasons....

let's see if i can implement quickly https://docs.github.com/en/rest/reference/repos#collaborators

**Comment by chmouel** [comment]
https://github.com/chmouel/tekton-asa-code/pull/108

**Comment by chmouel** [comment]
can you try to amend and repush --force to see if it kicks it ?

**Comment by rupalibehera** [comment]
@chmouel looks like there is some issue here it is not kicking back the CI

**Comment by chmouel** [comment]
There was an issue with this : 

`67m     Normal    CreateContainerConfigError   taskrun/tektoncd-operator-run-test-finally-kkz24         build step "step-send-to-slack" is pending with reason "secret \"slack-tektoncd-operator-ci-webhook\" not found"`

I needed to fix to the reinstall script, 


which could see in the console on clicking this link : 



It's an issue with my auto reinstal cron, I have fixed it manually..

**Comment by chmouel** [comment]
the lint task is going to fail tho : 

image

**Comment by chmouel** [comment]
and then it fails :-P 

feel free to iterate from there.

File: .tekton/pipeline.yaml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+                    fieldPath: metadata.labels['tekton.dev/pipelineRun']
+              - name: GITHUB_PULL_LABEL
+                value: "{{pull_request.labels}}"
+              - name: LABEL_TO_CHECK
+                value: "nightly-ci"
+              - name: SUCCESS_URL_ICON
+                value: "https://github.com/tektoncd.png"
+              - name: FAILURE_URL_ICON
+                value: "https://www.vhv.rs/dpng/d/415-4154815_grumpy-cat-png-photos-grumpy-cat-png-transparent.png"
+              - name: SUCCESS_SUBJECT
```
Comment (chmouel) [inline .tekton/pipeline.yaml:71]: any chances to find some other icons url so we can easily make a difference with the pipelines-catalog ci ?

Comment (rupalibehera) [reply .tekton/pipeline.yaml:71]: yep, I can change those

**Comment by rupalibehera** [comment]
> and then it fails :-P
> 
> feel free to iterate from there.



> and then it fails :-P
> 
> feel free to iterate from there.

It is going to fail because master branch does not contain any go code, it just contains openshift specifc stuff, I will just run these locally and check if everything is working and merge it, so that it runs on nightly PR's and other PR's made for releases and not run if a PR is against master branch.

**Comment by chmouel** [comment]
/lgtm
/approve

🚀

**Pull request merged**